tmt/base.py

```diff
@@ -210,7 +210,13 @@ def export(self, format_='yaml', keys=None):
         data['path'] = self.path
         if self.duration is not None:
             data['duration'] = self.duration
-        if self.environment is not None:
+        # Combine environment variables (plan overrides test)
+        if self.environment is not None or environment is not None:
+            combined_environment = dict()
+            if self.environment:
+                combined_environment.update(self.environment)
+            if environment:
+                combined_environment.update(environment)
```

Comment:
@psss  What about debug output of the combined environment, so user has change to understand what happened here for troubleshooting?

Reply:
ok, we can do that later :D